### PR TITLE
New feature: Added "Newsletter Subscription Form" widget

### PR DESCRIPTION
**Pull request by empiricompany**
### Description (*)

Add a new widget with newsletter subscription form.
It is useful for me to be able to insert the newsletter subscription form into CMS pages or Static Blocks.
I have always manually inserted this piece of code:
`{{block type="newsletter/subscribe" template="newsletter/subscribe.phtml"}}`

With the widget become:
{{widget type="newsletter/widget_subscribe" template="newsletter/subscribe.phtml"}}

But is the same, if you look the new block widget subscribe only extend block subscribe
```php
class Mage_Newsletter_Block_Widget_Subscribe extends Mage_Newsletter_Block_Subscribe implements Mage_Widget_Block_Interface
```

### Manual testing scenarios (*)

1. Insert Newsletter Subscribe Form Widget
2. Test with cache if form_key correctly change

demo

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by ADDISON74** [comment]
This is a good idea. Maybe we can extend it into another PR to the contact form too.

File: app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Newsletter
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (https://www.magento.com)
+ * @copyright  Copyright (c) 2020-2023 The OpenMage Contributors (https://www.openmage.org)
```
Comment (pquerner) [inline app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:12]: 2024

File: app/code/core/Mage/Newsletter/etc/widget.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * OpenMage
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE_AFL.txt.
+ * It is also available at https://opensource.org/license/afl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Newsletter
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (https://www.magento.com)
+ * @copyright  Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
```
Comment (pquerner) [inline app/code/core/Mage/Newsletter/etc/widget.xml:13]: 2024

File: app/locale/en_US/Mage_Newsletter.csv [hunk 3]
```diff
@@ -140,3 +140,4 @@
 "You have been unsubscribed.","You have been unsubscribed."
 "Your subscription has been confirmed.","Your subscription has been confirmed."
 "This email address is already confirmed.","This email address is already confirmed."
+"Newsletter Subscribe Form","Newsletter Subscribe Form"
```
Comment (fballiano) [inline app/locale/en_US/Mage_Newsletter.csv:143]:
```suggestion
"Newsletter Subscribe Form","Newsletter Subscribe Form"

```

just adding the new line, sorry

File: app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * @method string getSuccessMessage()
+ * @method string getErrorMessage()
+ * @method string getFormActionUrl()
+ */
+class Mage_Newsletter_Block_Widget_Subscribe extends Mage_Newsletter_Block_Subscribe implements Mage_Widget_Block_Interface
+{
+    protected function _construct()
+    {
+        parent::_construct();
+    }
```
Comment (fballiano) [inline app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:30]:
```suggestion
```

is the constructor needed since it just calls the parent?

Comment (fballiano) [reply app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:30]: I tested the PR without this constructor and it works fine

Comment (empiricompany) [reply app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:30]: I had kept it for safety to ensure the same behavior as the block if something is added in the future, but it's not that important

Comment (fballiano) [reply app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:30]: If I remember correctly we had a PR some months ago removing these kinds of constructors, but I see your point, let's think about that for a minute and see

**Comment by fballiano** [comment]
I would prefer not having the constructor only calling parent constructor, but I'll merge it anyway in a couple of days if we don't get more feedback

File: app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php (hunk 4 above)
Comment (mattdavenport) [reply app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:30]: I would vote to remove an 'empty' constructor method like this, especially if that is consistent with the rest of the code base. If we need to modify anything here in the future it should be simple enough to add back in.

Comment (luigifab) [reply app/code/core/Mage/Newsletter/Block/Widget/Subscribe.php:30]: drop it

**Review by ADDISON74 — APPROVED** [review]
I tested the PR. It adds to the list the newsletter widget that I used it on a CMS page. In the Frontend the form is displayed as expected and I successfully subscribed.

**Comment by ADDISON74** [comment]
@elidrissidev, @kiatng - please stamp this PR with your green color.

**Review by elidrissidev — APPROVED** [review]

File: app/code/core/Mage/Newsletter/etc/widget.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        <parameters>
+            <template translate="label">
+                <label>Template</label>
+                <visible>1</visible>
+                <type>select</type>
+                <value>newsletter/subscribe.phtml</value>
+                <values>
+                    <default translate="label">
+                        <value>newsletter/subscribe.phtml</value>
+                        <label>Newsletter Form Default Template</label>
```
Comment (kiatng) [inline app/code/core/Mage/Newsletter/etc/widget.xml:31]: Is this better:
```suggestion
                        <label>Default Template</label>
```
To be consistent with existing usage:
https://github.com/OpenMage/magento-lts/blob/aae923229112b80e7cf1e0b1d713e858efcd97f1/app/code/core/Mage/Sales/etc/widget.xml#L35-L38

Line 29: `<default translate="label">`,  need to add the label to Mage_Newsletter.csv if it's not there already.

File: app/code/core/Mage/Newsletter/etc/widget.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ * @category   Mage
+ * @package    Mage_Newsletter
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (https://www.magento.com)
+ * @copyright  Copyright (c) 2024 The OpenMage Contributors (https://www.openmage.org)
+ * @license    https://opensource.org/licenses/afl-3.0.php Academic Free License (AFL 3.0)
+ */
+-->
+<widgets>
+    <newsletter_subscribe type="newsletter/widget_subscribe" translate="name description" module="newsletter">
+        <name>Newsletter Subscribe Form</name>
```
Comment (kiatng) [inline app/code/core/Mage/Newsletter/etc/widget.xml:19]: I'm not a native speaker, it's usually "Newsletter Subscription Form". Or it's ok as is?

**Review by kiatng — APPROVED** [review]

**Review by mattdavenport — APPROVED** [review]

**Review by elidrissidev — APPROVED** [review]

**Pull request merged**